document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoQueryExecutor.java

```diff
@@ -80,7 +79,14 @@ public MongoCursor<BasicDBObject> aggregate(final Query originalQuery) {
             .collect(Collectors.toList());
 
     logPipeline(pipeline);
-    AggregateIterable<BasicDBObject> iterable = collection.aggregate(pipeline);
+
+    AggregateIterable<BasicDBObject> iterable;
```

Comment:
nit: As iterable is not used in the exception block, shall we do as below so if it's not null?
```
try {
      AggregateIterable<BasicDBObject> iterable = collection.aggregate(pipeline);
      return iterable.cursor();
    } catch (final MongoCommandException e) {
      log.error("Execution failed for query: {}. Aggregation Pipeline: {}", query, pipeline);
      throw e;
    }
 ```